### PR TITLE
Use Redis for the scheduled push queue

### PR DESCRIPTION
**Pull request by grahamgilbert**

### CODE REVIEW

File: director/schedule_push.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	counter := 0
+	for {
+		if !DevicesFetchedFromMDM {
+			time.Sleep(30 * time.Second)
+			log.Info("Devices are still being fetched from MicroMDM")
+			counter++
+			if counter > 10 {
+				break
+			}
+		} else {
```
Comment (discentem) [inline director/schedule_push.go:41]: Do we need an else here? Why not just break?

File: director/schedule_push.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+		err := processScheduledCheckin(pushQueue, task)
+		if err != nil {
+			ErrorLogger(LogHolder{Message: err.Error()})
+		}
+	}
+
+	fn()
+
+	for range ticker.C {
+		go fn()
```
Comment (discentem) [inline director/schedule_push.go:58]: Any reason to not do this function inline with the go routine?

Comment (grahamgilbert) [reply director/schedule_push.go:58]: Just because I think this is cleaner as it’s a pretty long function.

File: director/schedule_push.go (hunk 1 above)
Comment (grahamgilbert) [reply director/schedule_push.go:41]: I think we do? We wait for a variable set elsewhere to be true and then break

Comment (discentem) [reply director/schedule_push.go:41]: Yeah, what you are saying makes sense. I now understand the logic.

**Pull request merged**